FIX #22110: Skill Editor UI enhanced for both read only mode and read write mode

NITISH084 opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes https://github.com/oppia/oppia/issues/22110.
2. This PR does the following: 
- Moves the non-editable skill description block (*ngIf="!canEditSkillDescription()") into its own ```<div>``` placed after the editable input section for clearer visual and logical structure.

- Replaces the previous ```<p>``` wrapper (which incorrectly contained block elements) with a ```<div>``` to ensure valid HTML and consistent rendering.

- Adds conditional spacing between the label and the input/description using a ```<br />``` only for editable users.

- Removes the warning message for users who cannot edit the skill description.

- Shows the “Skill description should be at most 100 characters” warning only for users who can edit.

- Adjusts spacing between the italic text and the skill description in read-only mode to match the style of the “review material” section.

4. (For bug-fixing PRs only) The original bug occurred because: The skill description in read-only mode was similar to the editable mode and the layout had invalid HTML (block elements inside ```<p>```), causing rendering inconsistencies.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [X] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).


<!--
If this PR affects production server data, please follow
[these instructions](https://github.com/oppia/oppia/wiki/Testing-jobs-and-other-features-on-production#submitting-a-pr-with-a-new-job-or-feature-that-requires-third-party-api)
and link to the job request doc here.

## Proof that changes are correct

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->
### Before
<img width="1839" height="977" alt="image" src="https://github.com/user-attachments/assets/06a80407-52ad-48d0-812f-c3d601b42c62" />
Mobile view 
<img width="506" height="909" alt="image" src="https://github.com/user-attachments/assets/d9ad8bd9-c800-4929-8872-ef27e9126889" />


### AFTER
Read-only mode:
<img width="751" height="480" alt="Image" src="https://github.com/user-attachments/assets/cfdc8596-caa2-4f55-ba5a-f795dbb6a352" />

Read-write mode:
<img width="703" height="485" alt="Image" src="https://github.com/user-attachments/assets/a15ef60d-6c2d-428a-99f0-8645682ad3ca" />

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->
### Read-only mode:
<img width="1913" height="1079" alt="image" src="https://github.com/user-attachments/assets/9e216eb0-3685-40de-9184-45a17d8da775" />

### Read-write mode:
<img width="1890" height="914" alt="image" src="https://github.com/user-attachments/assets/a59872b3-9265-46a7-a044-110e0c158894" />


#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->
<img width="914" height="871" alt="image" src="https://github.com/user-attachments/assets/75a3f083-954e-4654-9842-bb96bf63065d" />


#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

Commit:
updated  core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

NITISH084 commented:
The CI test failing is already reported in issue https://github.com/oppia/oppia/issues/23141
Hey @amyyeung17 , Could you please re-run the CI tests? Thanks!

amyyeung17 commented on core/templates/pages/skill-editor-page/editor-tab/skill-description-editor/skill-description-editor.component.html:
```diff
@@ -19,8 +19,9 @@
-      <span *ngIf="!canEditSkillDescription()">{{ tmpSkillDescription }}</span>
-    </p>
+    </div>
+    <br *ngIf="canEditSkillDescription()">
```
If we are trying to add spacing between elements, perhaps we could use conditional styling instead of `<br>`? `<br>` are typically used for breaks within text. What do you think?

NITISH084 replied:
I initially added ```<br>``` for line breaks (similar to line 7), but I agree that using conditional styling would be a cleaner and more consistent solution. I’ll update it accordingly.

NITISH084 replied:
Done

amyyeung17 replied:
Would it be possible to use conditional margin styling? (e.g, margin-bottom: 1em or margin-top: 1 em)

NITISH084 replied:
<span class="line-break"  [ngStyle]="{
        'margin-bottom': canEditSkillDescription() ? '1em' : '0',
        'display': canEditSkillDescription() ? 'block' : 'none'
      }" ></span>

Is this ok?

amyyeung17 replied:
We might be able to simplify this a bit more. What about using a ternary statement for classes? That way, we would only need to show the condition once.

NITISH084 replied:
Yes It would be better, changed to ternary statements for classes.
```
<div [ngClass]="{'skill-editor-gap': canEditSkillDescription() }"></div>
 .skill-editor-gap {
    margin-bottom: 1em;
  }
```

NITISH084 replied:
> Yes It would be better, changed to ternary statements for classes.
> 
> ```
> <div [ngClass]="{'skill-editor-gap': canEditSkillDescription() }"></div>
>  .skill-editor-gap {
>     margin-bottom: 1em;
>   }
> ```
@amyyeung17  PTAL, Thanks!

amyyeung17 reviewed (changes requested):
Thanks for the descriptive overview. Just curious, was the text misaligned like that in the mobile version before your changes?

NITISH084 commented:
> Thanks for the descriptive overview. Just curious, was the text misaligned like that in the mobile version before your changes?

@amyyeung17 I’ve added screenshots of both the desktop and mobile views before my changes in the PR description for reference.

Commit:
replaced br tag with span and conditional styling

Commit:
resolved lint issues

amyyeung17 commented:
@NITISH084 Ohhh got it. Thanks!

Commit:
changed to ngstyle

Commit:
Updated skill-editoe-spacing

Commit:
updated the name of class

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

NITISH084 commented:
Failing CI tests , Issues Filed here
https://github.com/oppia/oppia/issues/22821
https://github.com/oppia/oppia/issues/23106
https://github.com/oppia/oppia/issues/23182

NITISH084 commented:
Hey  @HardikGoyal2003 , could you please rerun the failing tests when you get a chance and add this PR too in merge queue ?Thanks!

mon4our commented:
> > No, only do this when the changes you make actually change how the ui looks (like in this case) or introduce a new user journey. We do not follow this procedure for flaky tests.
> 
> @mon4our , could you please check out this issue related to my another https://github.com/oppia/oppia/actions/runs/17162864504/job/48765884353?pr=23133 PR, it also changes the UI, and as you said I tried to follow those steps but it shows this log in
> 
> ```
> With the provided path, there will be 1 file uploaded
> Artifact name is valid!
> Root directory input is valid!
> Error: Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run
> ```
> 
> Do you know how I should resolve this?
> 
> Thanks!

<img width="1152" height="1094" alt="image" src="https://github.com/user-attachments/assets/0b8ec2ad-2d9c-41fe-b6b4-c9123d6ec717" />

As both the desktop and mobile tests are failing, the desktop diff gets uploaded first. This prevents the mobile diff from getting uploaded as the artifacts share the same name. Try to fix the desktop screenshot first and then the mobile diff will appear.

I'll move this discussion to relevant PR. Also, in the future, if you need help, I suggest keeping the discussion in the PR with the context so that other contributors do not get confused when referring to it.

Commit:
updated image for desktop UI change

NITISH084 committed:
Merge branch 'develop' into Skill-editor-look-enhanced-for-both-read-only-mode-and-read-write-mode

Commit:
updated for desktop-dev too

Commit:
Changed the png for mobile acceptance test

NITISH084 commented:
@HardikGoyal2003 , could you please rerun the failing CI test , Issue filed here : https://github.com/oppia/oppia/issues/23189

imchristie reviewed (commented):
Why didn't you also update the screenshot `wokedExampleLimitError` in the folder prod-mobile-screenshots?

NITISH084 commented:
@imchristie I tried running the failing test on my local dev environment, and it was only failing for the dev version. I didn’t update the prod version yet , should I update it too?

Commit:
Revert "Removed dev screenshots"

This reverts commit 4900c4f839dbcd288656684a5fe643ba91fca47e.

Commit:
Revert "updated for desktop-dev too"

This reverts commit 8ee880e524bac4733e49c8f9424619150d5d1e4b.

Commit:
updated diff screenshot for prod mobiles

NITISH084 commented:
@imchristie  I have removed the dev screenshots and now added only the prod screenshots , i looked upon this https://github.com/oppia/oppia-web-developer-docs/pull/469 and followed it.

NITISH084 commented:
PTAL @imchristie ,Thanks.

imchristie reviewed (approved):
LGTM

NITISH084 commented:
Flakes filed https://github.com/oppia/oppia/issues/23278

NITISH084 commented:
Flake filed https://github.com/oppia/oppia/issues/21200  https://github.com/oppia/oppia/issues/16640
